### PR TITLE
Instrument the Prometheus Stats Exporter with trace

### PR DESCRIPTION
**Pull request by songy23**
Closes https://github.com/census-instrumentation/opencensus-java/issues/967.

A screenshot of the tracez page:
osvwlvaarfa

### CODE REVIEW

File: exporters/stats/prometheus/src/main/java/io/opencensus/exporter/stats/prometheus/PrometheusStatsCollector.java [hunk 1]
```diff
@@ -63,27 +72,49 @@ public static void createAndRegister() {
...
       }
+      span.addAnnotation("Finish collecting Prometheus Metric Samples.");
     } catch (Throwable e) {
       logger.log(Level.WARNING, "Exception thrown when collecting metric samples.", e);
+      span.setStatus(
+          Status.UNKNOWN.withDescription(
+              "Exception thrown when collecting Prometheus Metric Samples: "
+                  + exceptionMessage(e)));
     }
+    span.end();
```
Comment (bogdandrutu) [inline exporters/stats/prometheus/src/main/java/io/opencensus/exporter/stats/prometheus/PrometheusStatsCollector.java]: This should be moved in a finally block.

File: exporters/stats/prometheus/src/main/java/io/opencensus/exporter/stats/prometheus/PrometheusStatsCollector.java [hunk 2]
```diff
@@ -63,27 +72,49 @@ public static void createAndRegister() {
...
         samples.add(PrometheusExportUtils.createDescribableMetricFamilySamples(view));
       }
+      span.addAnnotation("Finish describing Prometheus Metrics.");
     } catch (Throwable e) {
       logger.log(Level.WARNING, "Exception thrown when describing metrics.", e);
+      span.setStatus(
+          Status.UNKNOWN.withDescription(
+              "Exception thrown when describing Prometheus Metrics: " + exceptionMessage(e)));
     }
+    span.end();
```
Comment (bogdandrutu) [inline exporters/stats/prometheus/src/main/java/io/opencensus/exporter/stats/prometheus/PrometheusStatsCollector.java]: same.

File: exporters/stats/prometheus/src/main/java/io/opencensus/exporter/stats/prometheus/PrometheusStatsCollector.java (hunk 1 above)
Comment (songy23) [reply exporters/stats/prometheus/src/main/java/io/opencensus/exporter/stats/prometheus/PrometheusStatsCollector.java]: Done.

File: exporters/stats/prometheus/src/main/java/io/opencensus/exporter/stats/prometheus/PrometheusStatsCollector.java (hunk 2 above)
Comment (songy23) [reply exporters/stats/prometheus/src/main/java/io/opencensus/exporter/stats/prometheus/PrometheusStatsCollector.java]: Done.

**Comment by songy23** [comment]
Thanks!

**Pull request merged**